enclave-runtime/src/lib.rs

```diff
@@ -448,6 +449,50 @@ pub unsafe extern "C" fn init_light_client(
 		Ok(header) => write_slice_and_whitespace_pad(latest_header_slice, header.encode()),
 		Err(e) => return e.into(),
 	}
+
+	// Initialize the global parentchain block import dispatcher instance.
+	let signer = match Ed25519Seal::unseal() {
+		Ok(s) => s,
+		Err(e) => {
+			error!("Error retrieving signer key pair: {:?}", e);
+			return sgx_status_t::SGX_ERROR_UNEXPECTED
+		},
+	};
+	let shielding_key = match Rsa3072Seal::unseal() {
+		Ok(s) => s,
+		Err(e) => {
+			error!("Error retrieving shielding key: {:?}", e);
+			return sgx_status_t::SGX_ERROR_UNEXPECTED
+		},
+	};
+
+	let validator_access = Arc::new(EnclaveValidatorAccessor::default());
+	let genesis_hash =
+		match validator_access.execute_on_validator(|v| v.genesis_hash(v.num_relays())) {
+			Ok(g) => g,
+			Err(e) => {
+				error!("Error retrieving genesis hash: {:?}", e);
+				return sgx_status_t::SGX_ERROR_UNEXPECTED
+			},
+		};
+
+	let stf_executor =
+		Arc::new(StfExecutor::new(Arc::new(OcallApi), Arc::new(GlobalFileStateHandler)));
+	let extrinsics_factory =
+		Arc::new(ExtrinsicsFactory::new(genesis_hash, signer, GLOBAL_NONCE_CACHE.clone()));
+	let indirect_calls_executor =
+		Arc::new(IndirectCallsExecutor::new(shielding_key, stf_executor.clone()));
+	let parentchain_block_importer = Arc::new(ParentchainBlockImporter::new(
+		validator_access,
+		Arc::new(OcallApi),
+		stf_executor,
+		extrinsics_factory,
+		indirect_calls_executor,
+	));
+	let block_import_dispatcher = Arc::new(ImmediateDispatcher::new(parentchain_block_importer));
+
+	GLOBAL_DISPATCHER_COMPONENT.initialize(block_import_dispatcher);
```

Comment:
We initialize the dispatcher once, here. We can then just access it whenever we get an e-call that requires it (and thus save all the construction work, including reading signer, shielding key and genesis hash.

Reply:
wuhuw, that's great!